homeassistant/components/mold_indicator/config_flow.py

```diff
@@ -0,0 +1,115 @@
+"""Config flow for Mold Indicator integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN, SensorDeviceClass
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_NAME
+from homeassistant.core import DOMAIN as HOMEASSISTANT_DOMAIN
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import selector
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+
+from .const import (
+    CONF_CALIBRATION_FACTOR,
+    CONF_INDOOR_HUMIDITY,
+    CONF_INDOOR_TEMP,
+    CONF_OUTDOOR_TEMP,
+    DEFAULT_NAME,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+ISSUE_PLACEHOLDER = {"url": "/config/integrations/dashboard/add?domain=mold_indicator"}
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_INDOOR_TEMP): selector.EntitySelector(
+            selector.EntitySelectorConfig(
+                domain=SENSOR_DOMAIN,
+                filter=selector.EntityFilterSelectorConfig(
+                    device_class=SensorDeviceClass.TEMPERATURE
+                ),
+            ),
+        ),
+        vol.Required(CONF_OUTDOOR_TEMP): selector.EntitySelector(
+            selector.EntitySelectorConfig(
+                domain=SENSOR_DOMAIN,
+                filter=selector.EntityFilterSelectorConfig(
+                    device_class=SensorDeviceClass.TEMPERATURE
+                ),
+            ),
+        ),
+        vol.Required(CONF_INDOOR_HUMIDITY): selector.EntitySelector(
+            selector.EntitySelectorConfig(
+                domain=SENSOR_DOMAIN,
+                filter=selector.EntityFilterSelectorConfig(
+                    device_class=SensorDeviceClass.HUMIDITY
+                ),
+            ),
+        ),
+        vol.Optional(CONF_CALIBRATION_FACTOR): vol.Coerce(float),
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    }
+)
+
+
+class MoldIndicatorConfigFlow(ConfigFlow, domain=DOMAIN):
```

Comment:
Consider deriving from `SchemaConfigFlowHandler`, that makes it much easier to add an option flow with pre-filled values and so on.

Reply:
I've just had a quick look, and there doesn't seem to be much else using this (other than built-in components). Am I missing something? 🤔

Reply:
> I've just had a quick look, and there doesn't seem to be much else using this (other than built-in components). Am I missing something? 🤔

Currently, that is true. But I am working on new integrations with schema config flows and I like them very much 😄

Reply:
Locally I've managed to get this working nicely. The only thing I'm hitting now is that when modifying the integration entity at all (e.g. changing the calibration factor), it becomes unavailable.

I'm thinking this is because of the use of `mold_indicator_startup` (which runs once on the `EVENT_HOMEASSISTANT_START` event), but not sure what to do about this. 🤔

Reply:
Why are we waiting for `EVENT_HOMEASSISTANT_START`, is it to wait for sensors we depend on to be added to the state machine?